Issue 156 (#505)
SwiftWeekly/swiftweekly.github.io: _drafts/2020-03-26-issue-156.md

---
layout: post
title: ! 'Issue #156'
author: fassko
---

> TODO: intro comments

<!--excerpt-->

{% include indie_support_weeks.html link="https://outercorner.com/blog/2019/05/secrets-3.0/" heading="Secrets" body="Protect your secrets. Easily and securely store passwords, credit card numbers, bank details, and confidential information with Secrets." displaylink="outercorner.com" %}

### Starter tasks

> TODO

### Podcasts

> TODO: Latest episode(s) of Swift Unwrapped & the Swift Community Podcast

### News and community

> Apple has [announced](https://developer.apple.com/wwdc20/) WWDC 2020 and it will be a completely new online experience.

Reviewer: ```suggestion
Apple [announced](https://developer.apple.com/wwdc20/) WWDC 2020 will be a completely online experience this year.
```

---
layout: post
title: ! 'Issue #156'
author: fassko
---

> Apple announced that WWDC 2020 will be an online-only event, and many of us are learning how to work remotely from our homes while we cope with the going on health crisis. I hope everyone is staying safe.

Reviewer: ```suggestion
Apple announced that WWDC 2020 will be an online-only event, and many of us are learning how to work remotely from our homes while we cope with the on-going health crisis. I hope everyone is staying safe.
```

oops. this was my bad 😜

---
layout: post
title: ! 'Issue #156'
author: fassko
---

> It seems that the world has turned upside down because of the Covid-19 virus. Most of the countries have been shut down and there is a lot of unknown in the future.

Reviewer: I think we should remove this first sentence. I know it's tempting to mention, but I think it's better to mostly stay on topic. (see suggestion below) Plus, I think folks are getting overwhelmed, constantly hearing about this. 😊

Reviewer 2: I agree, although I still want to keep it. :P

> Apple has announced that WWDC 2020 will be a remote event. We all are learning how to work remotely from our homes to cope with all the madness going on.

Reviewer: ```suggestion
Apple announced that WWDC 2020 will be an online-only event, and many of us are learning how to work remotely from our homes while we cope with the going on health crisis. I hope everyone is staying safe.
```

> Meanwhile, Swift 5.2 has been officially released along alongside with Xcode 11.4.

Reviewer: ```suggestion
Meanwhile, Swift 5.2 has been officially released with Xcode 11.4. So take a break from the news and spend some time updating your projects!
```

### Starter tasks

> TODO

### Podcasts

In [episode 69](https://www.swiftbysundell.com/podcast/69/) of Swift by Sundell podcast [Holly Borla](https://twitter.com/hollyborla) and [Grace Kendall](https://twitter.com/gracekendall26), both software engineers at Apple, join [John Sundell](https://twitter.com/johnsundell) to go on a deep dive into the Swift Playgrounds app and Swift 5.2’s new diagnostics engine.

### News and community

Apple [announced](https://developer.apple.com/wwdc20/) WWDC 2020 will be a completely online experience this year.

Apple [released](https://swift.org/blog/swift-5-2-released/) Swift 5.2 and [announced](https://developer.apple.com/documentation/xcode_release_notes/xcode_11_4_release_notes) Xcode 11.4.

[Nicole Jacque](https://github.com/najacque/) wrote a [blog post](https://swift.org/blog/5-3-release-process/) about Swift 5.3 release process.

[Paul Hudson](https://twitter.com/twostraws) wrote a great [article](https://www.hackingwithswift.com/articles/212/whats-new-in-swift-5-2) explaining what's new in Swift 5.2.

> [Federico Zanetello](https://twitter.com/zntfdr) wrote two great articles about [ArgumentParser](https://www.fivestars.blog/code/a-look-into-argument-parser.html) and [Swift Executables Progress State](https://www.fivestars.blog/code/executables-progress.html).

Reviewer: ```suggestion
[Federico Zanetello](https://twitter.com/zntfdr) wrote two great articles, one about the new [ArgumentParser](https://www.fivestars.blog/code/a-look-into-argument-parser.html) and another about [Swift Executables Progress State](https://www.fivestars.blog/code/executables-progress.html).
```

> He did not stop there are created a [presentation](https://speakerdeck.com/zntfdr/whats-new-in-swift-5-dot-2) about new features in Swift 5.2.

Reviewer: ```suggestion
He also created a [presentation](https://speakerdeck.com/zntfdr/whats-new-in-swift-5-dot-2) about new features in Swift 5.2.
```

> Great [talk](https://drive.google.com/file/d/1gI6Zk2jS0-MNkckYBnRFNtVCHoGKHWct/view) about Swift for TensorFlow which should have been given at TF Dev Summit.

Reviewer: ```suggestion
A great [talk](https://drive.google.com/file/d/1gI6Zk2jS0-MNkckYBnRFNtVCHoGKHWct/view) was shared about Swift for TensorFlow which would normally have been given at TF Dev Summit.
```

> RayWenderlich folks wrote a great [tutorial](https://www.raywenderlich.com/8016626-swiftnio-tutorial-practical-guide-for-asynchronous-problems) on how to start with SwiftNIO.

Reviewer: ```suggestion
The Ray Wenderlich folks wrote a great [tutorial](https://www.raywenderlich.com/8016626-swiftnio-tutorial-practical-guide-for-asynchronous-problems) on how to get started with SwiftNIO.
```

> [Bruno Rocha](https://twitter.com/rockbruno_) [wrote](https://swiftrocks.com/useful-global-swift-functions.html) an article explaining to us about useful global swift functions.

Reviewer: ```suggestion
[Bruno Rocha](https://twitter.com/rockbruno_) [wrote](https://swiftrocks.com/useful-global-swift-functions.html) an article about useful global swift functions.
```

### Swift Forums

[Maksim Kita](https://github.com/kitaisreal) pitched a `Circular Buffer`.

> Introduce the `CircularBuffer` collection type conforming to the `RandomAccessCollection`,
`RangeReplaceableCollection` and `MutableCollection` protocols. With random element
access and support for constant back and front element insertion and deletion.
> 
> Swift currently does not have collection with both element random access
and constant O(1) elements back and front insertion and deletion. A good
usage examples for such collection are queue, deque, fixed length buffers.
Such abstractions cannot effectively be build on Array because of O(n) first element deletion.

[Aamir Nazir](https://forums.swift.org/u/aamir-nazir) opened discussion about counting occurrences of a substring in a string.

> Given the fact that Swift String class doesn't has any method for counting occurrences of a substring in a string. i.e
> Counting occurrences of `"Swift"` in `"Hello Swift Swift"` => 2
> 
> There are many alternatives available for counting occurrences of substring but I think it would be more intuitive if we add support of counting occurrences of a substring in a string.

> [Nate Cook](https://twitter.com/nnnnnnnn) [pitched](https://forums.swift.org/t/main-type-based-program-execution/34624) idea about adding a new attribute that you can use to designate a type that provides the entry point for a Swift program.

Reviewer: ```suggestion
[Nate Cook](https://twitter.com/nnnnnnnn) [pitched](https://forums.swift.org/t/main-type-based-program-execution/34624) an idea about adding a new attribute that you can use to designate a type that provides the entry point for a Swift program.
```

> This is a generalization of the `@UIApplicationMain` and `@NSApplicationMain` attributes that have been in Swift from the beginning, making that specialized behavior available to any library or framework.
> 
> The Swift compiler will recognize a type annotated with the `@main` attribute as providing the entry point for a program. Types marked with `@main` have a single implicit requirement: declaring a static `main()` method.

[tomerd](https://forums.swift.org/u/tomerd) [shared](https://forums.swift.org/t/march-4th-2020/34617) Swift on the Server Workgroup March 4th, 2020 meeting notes.

> [Karoy Lorentey](https://twitter.com/lorentey) [pitched](https://forums.swift.org/t/low-level-atomic-operations/34683) idea bout Low-Level Atomic Operations.

Reviewer: ```suggestion
[Karoy Lorentey](https://twitter.com/lorentey) [pitched](https://forums.swift.org/t/low-level-atomic-operations/34683) an idea bout Low-Level Atomic Operations.
```

>Here is a pitch for adding a limited set of low-level atomic operations to the Standard Library, including native spellings for C++-style acquire-release memory orderings. Our goal is to enable intrepid library authors to start building synchronization constructs directly in Swift.
> 
> This means we must start talking about how these things will work in Swift -- in other words, we need to start working on a concurrency memory model. Given that Swift already interoperates with the C/C++ memory model, it seems like a good idea to use that as a starting point.

> [Saleem Abdulrasool](https://twitter.com/compnerd) [showed](https://forums.swift.org/t/new-swift-installer-for-windows/34692) us the new Swift Installer for Windows.

Reviewer: ```suggestion
[Saleem Abdulrasool](https://twitter.com/compnerd) [shared](https://forums.swift.org/t/new-swift-installer-for-windows/34692) the new Swift Installer for Windows.
```

[Cal Stephens](https://forums.swift.org/u/cal) started a [discussion](https://forums.swift.org/t/codingkeypath-add-support-for-encoding-and-decoding-nested-objects-with-dot-notation/34710) about how to add a new CodingKeyPath type that allows consumers to key into nested objects using [dot notation](https://developer.apple.com/documentation/objectivec/nsobject/1416468-value).

>Today, encoding and decoding Codable objects using the compiler's synthesized implementation requires that your object graph has a one-to-one mapping to the object graph of the target payload. This decreases the control that authors have over their `Codable` models.
> 
> I propose that we add a new `CodingKeyPath` type that allows consumers to key into nested objects using [dot notation](https://developer.apple.com/documentation/objectivec/nsobject/1416468-value).

> [Hassan ElDesouky](https://forums.swift.org/u/HassanElDesouky) started [discussion](https://forums.swift.org/t/contribution-starter-guide/34747) about creating a Swift contribution starter guide. PReviously he [gathered](https://forums.swift.org/t/important-talks-and-articles-for-first-time-swift-contributors/34537) materials of important talks and articles for first time contributors.

Reviewer: ```suggestion
[Hassan ElDesouky](https://forums.swift.org/u/HassanElDesouky) started [discussion](https://forums.swift.org/t/contribution-starter-guide/34747) about creating a Swift contribution starter guide. Previously, he [gathered](https://forums.swift.org/t/important-talks-and-articles-for-first-time-swift-contributors/34537) materials of important talks and articles for first time contributors.
```

{% include indie_support_weeks.html link="https://sundialapp.com" heading="Sundial - Solar & Lunar Times and Alerts" body="Sundial provides essential sun and moon data in a compact, beautiful display. Create alerts for 24 solar/lunar events. Want to be notified 30 minutes before sunset to take the dog out for a walk? Piece of cake. Need an alert the day before full/new moon to take rest for Ashtanga yoga? No problem. For iPhone, iPad, and Apple Watch." displaylink="sundialapp.com" %}

### Finally

> A [way](https://twitter.com/jckarter/status/1238669170767585280) to print a comma-separated list.

Reviewer: ```suggestion
Here's a [great way](https://twitter.com/jckarter/status/1238669170767585280) to print a comma-separated list.
```
